File: merging/python/merge_single.py


def merge_single(array, low, mid, high):
  i = low
  j = mid + 1
  k = low
  result = [None] * len(array)
  while(i <= mid and j <= high):
    if array[i] < array[j]:
      result[k] = array[i]
      k+=1
      i+=1
    else:
      result[k] = array[j]
      j+=1
      k+=1
  
  while(i <= mid):
    result[k] = array[i]
    k+=1
    i+=1
  while(j <= high):
    result[k] = array[j]
    k+=1
    j+=1
  
  return result
  

num_list = [1, 4, 7, 2, 5, 6]
h = len(num_list) -1
m = h //2
print(merge_single(num_list, 0,m, h))

File: merging/js/merge_single.js
function mergeSingle(arr, low, mid, high) {
  let i = low;
  let j = mid + 1;
  let k = low;
  let temp = [];
  while (i <= mid && j <= high) {
    if (arr[i] < arr[j]) temp[k++] = arr[i++];
    else temp[k++] = arr[j++];
  }
  for (; i <= mid; i++) {
    temp[k++] = arr[i];
  }
  for (; j <= high; j++) {
    temp[k++] = arr[j];
  }
  return temp;
}
let x = [1, 4, 7, 2, 5, 6];
const high = x.length - 1;
const mid = ~~(high / 2);

console.log(mergeSingle(x, 0, mid, high));

function mergeList(arr, low, mid, high) {
  let i = 0;
  let j = mid + 1;
  let k = low;
  let temp = [];

  while (i <= mid && j <= high) {
    if (arr[i] < arr[j]) {
      temp[k++] = arr[i++];
    } else temp[k++] = arr[j++];
  }
  for (; i <= mid; i++) {
    temp[k++] = arr[i];
  }
  for (; j <= high; j++) {
    temp[k++] = arr[j];
  }
  return temp;
}

console.log(mergeList(x, 0, mid, high));
